docs/features/streaming.rst

```diff
@@ -0,0 +1,100 @@
+.. _streaming:
+
+Response Streaming
+==================
+
+By using mitmproxy's streaming feature, response contents can be passed to the client incrementally
+before they have been fully received by the proxy. This is especially useful for large binary files
+such as videos, where buffering the whole file slows down the client's browser.
+
+Request Streaming
+=================
+
+As the name suggests, request streaming feature can be used to stream request body to the server incrementally
+before it has been fully received by the proxy. This may be useful for large file uploads.
+
+
+By default, mitmproxy will read the entire request/response, perform any indicated
+manipulations on it and then send the (possibly modified) request or response to
+the server or the client respectively. In some cases this is undesirable and you may wish to "stream"
+the request/response. When streaming is enabled, the request/response is
+not buffered on the proxy but directly sent to the server/client instead.
```

Comment:
Move this paragraph to the top of the file, underneath a top-level headline: **HTTP Streaming**
Also mention that HTTP headers are still fully buffered before being sent.

Then the two sub-headlines for **Request Body Streaming** and **Response Body Streaming**.